src/main/java/com/bandwidth/sdk/messaging/exception/MessagingClientException.java

```diff
@@ -1,17 +1,34 @@
 package com.bandwidth.sdk.messaging.exception;
 
 
+import java.util.Optional;
+
 public class MessagingClientException extends RuntimeException {
+    private final Optional<Integer> statusCode;
 
     public MessagingClientException(String message) {
         super(message);
+        statusCode = Optional.empty();
     }
 
     public MessagingClientException(Throwable cause) {
         super(cause);
+        statusCode = Optional.of(cause)
+                .filter(throwable -> throwable instanceof MessagingClientException)
+                .flatMap(throwable -> ((MessagingClientException) throwable).getStatusCode());
     }
 
     public MessagingClientException(String message, Throwable cause) {
         super(message, cause);
+        statusCode = Optional.empty();
```

Comment:
Why isn't the statusCode parsed from the throwable in this constructor?

Reply:
Okay I added that logic to that constructor. Why does this constructor exist? It is not used anywhere